src/messages.rs

```diff
@@ -0,0 +1,299 @@
+use bytes::BytesMut;
+use secp256k1::key::PublicKey;
+use secp256k1::Secp256k1;
+use secp256k1::Signature;
+use std::io;
+use tokio_io::codec::{Encoder, Decoder};
+
+/// Network messages take the following form:
+///
+/// | Field   | Length | Description                                                   |
+/// |---------|--------|---------------------------------------------------------------|
+/// | Magic   | 4      | Fixed magic bytes for identifying network messages            |
+/// | Header  | 1      | Message header                                                |
+/// | Payload | ...    | Message payload                                               |
+///
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub struct Message {
+    header: Header,
+    payload: Payload,
+}
+
+impl Message {
+    fn new(header: Header, payload: Payload) -> Message {
+        Message {
+            header: header,
+            payload: payload,
+        }
+    }
+    /// Takes a byte slice and builds a Message. Returns same as MessageCodec::decode plus the
+    /// remaining (unconsumed) buffer.
+    fn from_bytes<'a>(
+        buf: &'a [u8],
+        secp256k1: &Secp256k1,
+    ) -> io::Result<Option<(Message, &'a [u8])>> {
+        // Magic
+        if buf.len() < 4 {
+            return Ok(None);
+        }
+        let (magic, buf) = buf.split_at(4);
+        if !magic.starts_with(&[0x68, 0x1e, 0x58, 0x12]) {
```

Comment:
Can we have the magic bytes as a constant? Also, a simple equality  check will be clearer I think